### ISSUE TITLE
What Front-end framework to use?

### ISSUE THREAD

**Issue by lcabrini** [AluguntuguiProject/aluguntugui#15]
Pardon my ignorance here, but I'm not sure what RoR supports "out of the box" and what it doesn't. We need to decide on what we are going to support for the front-end development, if anything. Ideas here are:

1. nothing. That is, do it all ourselves
2. whatever RoR supports, which TIMBW, jquery
3. Bootstrap, loved by everybody
4. angular.js, set to take over the world, or so I hear
5. something else, as I said, I am largely ignorant in this area

What are your thoughts on this? I am really not opinionated here.

**Comment by benbaka** [comment]
I would suggest that we rather work out something till we get to the point where some of these tools are needed. 

Just for design purposes we will use Bootstrap. 

Some gems will become necessary naturally as we setup some fields on models. But I would not go in for too many gems right not else there will be nothing to learn.

**Comment by lcabrini** [comment]
Good point. So we have agreed to use Bootstrap at this current time, right? Is it supported by RoR out-of-the-box or we need to install it manually?

**Comment by benbaka** [comment]
For bootstrap let's use the bootstrap gem.

**Comment by lcabrini** [comment]
Ok, cool. So what next? Create an issue for this dep to be added?

**Comment by benbaka** [comment]
Well, you can create an issue associated with the user story for which you need the gem. That would give it the proper context.

**Comment by lcabrini** [comment]
Yes, that sounds reasonable. There is no need to bring in any dependency until we need it, right? And when that time comes we make it one of the dependent tasks of the user story that brings in the need for the framework.

**Comment by benbaka** [comment]
Yes, that sounds reasonable. There is no need to bring in any dependency
until we need it, right?
I agree.

And when that time comes we make it one of the dependent tasks of the user
story that brings in the need for the framework.
Cool.
